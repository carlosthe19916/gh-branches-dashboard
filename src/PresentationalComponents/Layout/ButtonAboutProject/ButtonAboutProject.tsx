import * as React from "react";
import {
  AboutModal,
  TextContent,
  TextList,
  TextListItem,
  Button,
  ButtonVariant
} from "@patternfly/react-core";
import { HelpIcon } from "@patternfly/react-icons";
import brandImage from "../../../logo.svg";

export interface ButtonAboutProjectProps {}

interface State {
  isOpen: boolean;
}

export class ButtonAboutProject extends React.Component<
  ButtonAboutProjectProps,
  State
> {
  constructor(props: ButtonAboutProjectProps) {
    super(props);
    this.state = {
      isOpen: false
    };
  }

  toggleButton = () => {
    this.setState(({ isOpen }) => ({
      isOpen: !isOpen
    }));
  };

  render() {
    const { isOpen } = this.state;

    return (
      <React.Fragment>
        <Button
          id="aboutButton"
          aria-label="About button"
          variant={ButtonVariant.plain}
          onClick={this.toggleButton}
        >
          <HelpIcon />
        </Button>
        <AboutModal
          isOpen={isOpen}
          onClose={this.toggleButton}
          trademark="COPYRIGHT Â© 2020."
          brandImageSrc={brandImage}
          brandImageAlt="Logo"
          productName="gh-branches-dashboard"
        >
          <TextContent>
            <TextList component="dl">
              <TextListItem component="dt">Source code</TextListItem>
              <TextListItem component="dd">
                <a href="https://github.com/carlosthe19916/gh-branches-dashboard">
                  Github
                </a>
              </TextListItem>
            </TextList>
          </TextContent>
        </AboutModal>
      </React.Fragment>
    );
  }
}
