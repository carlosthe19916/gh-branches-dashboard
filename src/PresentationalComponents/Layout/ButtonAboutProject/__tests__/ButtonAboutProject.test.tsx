import React from "react";
import { shallow } from "enzyme";
import { ButtonAboutProject } from "../ButtonAboutProject";

it("Test snapshot", () => {
  const wrapper = shallow(<ButtonAboutProject />);
  expect(wrapper).toMatchSnapshot();
});

it("Test snapshot :: open modal", () => {
  const wrapper = shallow(<ButtonAboutProject />);
  wrapper.find("#aboutButton").simulate("click");
  expect(wrapper).toMatchSnapshot();
});
