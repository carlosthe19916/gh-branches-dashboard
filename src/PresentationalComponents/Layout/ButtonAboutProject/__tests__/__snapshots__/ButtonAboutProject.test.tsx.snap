// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot :: open modal 1`] = `
<Fragment>
  <Component
    aria-label="About button"
    id="aboutButton"
    onClick={[Function]}
    variant="plain"
  >
    <HelpIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
      title={null}
    />
  </Component>
  <AboutModal
    appendTo={null}
    backgroundImageSrc=""
    brandImageAlt="Logo"
    brandImageSrc="logo.svg"
    className=""
    isOpen={true}
    noAboutModalBoxContentContainer={false}
    onClose={[Function]}
    productName="gh-branches-dashboard"
    trademark="COPYRIGHT © 2020."
  >
    <TextContent>
      <TextList
        component="dl"
      >
        <TextListItem
          component="dt"
        >
          Source code
        </TextListItem>
        <TextListItem
          component="dd"
        >
          <a
            href="https://github.com/carlosthe19916/gh-branches-dashboard"
          >
            Github
          </a>
        </TextListItem>
      </TextList>
    </TextContent>
  </AboutModal>
</Fragment>
`;

exports[`Test snapshot 1`] = `
<Fragment>
  <Component
    aria-label="About button"
    id="aboutButton"
    onClick={[Function]}
    variant="plain"
  >
    <HelpIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
      title={null}
    />
  </Component>
  <AboutModal
    appendTo={null}
    backgroundImageSrc=""
    brandImageAlt="Logo"
    brandImageSrc="logo.svg"
    className=""
    isOpen={false}
    noAboutModalBoxContentContainer={false}
    onClose={[Function]}
    productName="gh-branches-dashboard"
    trademark="COPYRIGHT © 2020."
  >
    <TextContent>
      <TextList
        component="dl"
      >
        <TextListItem
          component="dt"
        >
          Source code
        </TextListItem>
        <TextListItem
          component="dd"
        >
          <a
            href="https://github.com/carlosthe19916/gh-branches-dashboard"
          >
            Github
          </a>
        </TextListItem>
      </TextList>
    </TextContent>
  </AboutModal>
</Fragment>
`;
